src/kOS/Suffixed/VesselSensors.cs

```diff
@@ -36,29 +34,65 @@ private void FindSensors(Vessel target)
                 {
                     if (module is ModuleEnviroSensor)
                     {
-                        switch (module.Fields.GetValue("sensorType").ToString())
+                        var moduleSensorType = module.Fields.GetValue("sensorType").ToString();
+                            if (moduleSensorType != sensorType) continue;
+                        switch (moduleSensorType)
                         {
                             case "ACC":
-                                acceleration = new Vector(target.acceleration);
-                                break;
-                            case "PRES":
-                                pressure = target.staticPressurekPa;
-                                break;
-                            case "TEMP":
-                                temperature = part.temperature;
-                                break;
+                                return new Vector(vessel.acceleration);
                             case "GRAV":
-                                geeForce = new Vector(FlightGlobals.getGeeForceAtPosition(part.transform.position));
-                                break;
+                                return new Vector(FlightGlobals.getGeeForceAtPosition(part.transform.position));
+                                
                         }
                     }
-                    foreach (var c in part.FindModulesImplementing<ModuleDeployableSolarPanel>())
+                }
+            }
+            throw new KOSException("Cannot find sensor for " + sensorType);
+        }
+
+        private double GetSensorDoubleValue(string sensorType)
+        {
+            foreach (var part in vessel.Parts)
+            {
```

Comment:
This oarts walk is going to be a bit expensive if someone is trying to query the value over and over in a loop to see how it changes each physics tick.

Could you do the parts walk during the constructor, and save the set of PartModules of each type then?  That would be similar to what it did before, where it stored the values in private variables, but now it's just storing the PartModules rather than the actual values, deferring the query of the partmodules for later when you request a value.

That way you only pay the expense of this parts walk once, during the constructor, and after that you already have the relevant PartModules to query each time.

It does raise an issue with having stale parts that become wrong after a decouple or docking operation changes the parts list, but you can guard against that by checking if the PartModules's Part's Vessel is equal to the vessel you remembered in the constructor.  If it's not, then that means the PartModule is now on a different vessel, and that means you need to re-do the parts walk to re-obtain the PartModule list again.